src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -893,13 +894,18 @@ public async Task<ResultSetSubset> InterServiceResultSubset(SubsetParams subsetP
             // Setup the ResultSet updated callback
             ResultSet.ResultSetAsyncEventHandler resultUpdatedCallback = async r =>
             {
+                ExecutionPlanGraph graph = null;
+                if(r.Summary.Complete && r.Summary.SpecialAction.ExpectYukonXMLShowPlan == true)
+                {
+                    var xmlString = r.GetRow(0)[0].DisplayValue;
+                    graph = ShowPlanGraphUtils.CreateShowPlanGraph(xmlString);
+                }
                 ResultSetUpdatedEventParams eventParams = new ResultSetUpdatedEventParams
                 {
                     ResultSetSummary = r.Summary,
-                    OwnerUri = ownerUri
+                    OwnerUri = ownerUri,
+                    ShowPlanGraph = graph
```

Comment:
I'm concerned about sending over a huge showplan graph - have you tested if there's any perf issues with those?

Reply:
Since we are generating the show plan graph only for the last result set update, I don't think it will be a perf issue for the users. Also, most of the show plans graph are not excessively big.

Reply:
When it's generated doesn't really matter if the size is too large.

How are you coming to your statement "most of the show plan graph are not that excessively big"? I'm not sure that's something we can take for granted. I would suggest talking to the QP experts in that area to get a sense for whether this is something we should be more careful of. And at the very least you should be doing testing to see if there's a limit for when perf starts to become a noticeable issue.

Reply:
we should look at the perf of this scenario to understand it.  but we need to transmit the graph data to the client to render it at some point.  I guess maybe the XML is more compact than the processed JSON format. Though I suspect compared to the size of other things we pass around (e.g. query resultsets).  I suspect the delta between the XML format and JSON format isn't that impactful.